client/my-sites/checkout/checkout-thank-you/jetpack-checkout-siteless-thank-you.tsx

```diff
@@ -38,6 +38,11 @@ const JetpackCheckoutSitelessThankYou: FC< Props > = ( { productSlug, receiptId
 	const translate = useTranslate();
 	const dispatch = useDispatch();
 
+	const validUrl = ( url: string ): boolean => {
+		const pattern = new RegExp( '^https?:\\/\\/\\w+\\.[a-z]{2,}', 'i' );
+		return !! pattern.test( url );
+	};
```

Comment:
This regular expression concerns me because I don't understand it well enough to know we won't be blocking valid URLs. For instance, `http://required-clownfish.jurassic.ninja` and `required-clownfish.jurassic.ninja` both fail the test. It looks like it doesn't allow the `-` character.

My guess is we need something more "open".

Reply:
`@wordpress/url` [has a isURL method that seems](https://www.npmjs.com/package/@wordpress/url) like it would be a good fit?

Reply:
Good call. I think that might work – it basically uses `new URL( input )` under the hood. A minor annoyance is that users will have to enter the protocol; otherwise, it will reject the URL (maybe it is a good thing).

Let's use that function instead. 

Thank you for jumping in, @cbauerman!

Reply:
The only problem with `isUrl` is it allows:  'http://hi`  - it doesn't require a dot(.)  What do you think, are we ok with that?

When I found that out, I then was going to use a combination of `isUrl` and regex for a dot.  But then found that it may be just as easy to make the validUrl regex that I made.  But yes, I forgot to allow `-`.    Either way, whatever you think is best, works for me. 🙂

Reply:
> The only problem with isUrl is it allows: 'http://hi` - it doesn't require a dot(.) What do you think, are we ok with that?

Yes. Let's go with `isUrl` since we already use it in other parts of the codebase.